openhands/server/session/agent_session.py

```diff
@@ -231,15 +233,31 @@ async def _create_runtime(
                 )
             return
 
-        self.runtime.clone_repo(github_token, selected_repository)
-        if agent.prompt_manager:
-            agent.prompt_manager.load_microagent_files(
-                self.runtime.get_custom_microagents(selected_repository)
-            )
+        if self.runtime is not None:
+            # Add port mapping instructions based on runtime type
+            port_instructions = None
+            if isinstance(self.runtime, EventStreamRuntime):
+                port_instructions = (
+                    '* When serving some application, use ports 4141 and 4142.'
+                )
+            elif isinstance(self.runtime, RemoteRuntime):
+                port_instructions = (
+                    '* When serving some application, use ports 81623 and 81624.'
+                )
```

Comment:
This should be a method on the runtime, like `get_app_ports`, instead of using `isinstance`. Then we can have a template here like:
```
f'* If you need to run a web application, use one of these ports whenever possible: {ports.join(',')}'
```